core/src/main/java/org/elasticsearch/index/seqno/GlobalCheckpointTracker.java

```diff
@@ -641,17 +692,23 @@ public long clusterStateVersion() {
             return localCheckpoints;
         }
 
+        public IndexShardRoutingTable getRoutingTable() {
+            return routingTable;
+        }
+
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             out.writeVLong(clusterStateVersion);
             out.writeMap(localCheckpoints, (streamOutput, s) -> out.writeString(s), (streamOutput, lcps) -> lcps.writeTo(out));
+            IndexShardRoutingTable.Builder.writeTo(routingTable, out);
         }
 
         @Override
         public String toString() {
             return "PrimaryContext{" +
                     "clusterStateVersion=" + clusterStateVersion +
                     ", localCheckpoints=" + localCheckpoints +
+                    ", routingTable=" + routingTable +
```

Comment:
I changed the newly introduced `toString()` method on `IndexShardRoutingTable` to be single line so that we still get useful output.